src/common/libjob/unwrap.c

```diff
@@ -0,0 +1,116 @@
+/************************************************************\
+ * Copyright 2022 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <errno.h>
+#include <string.h>
+#include <stdint.h>
+#include <unistd.h>
+
+#if HAVE_FLUX_SECURITY
+#include <flux/security/context.h>
+#include <flux/security/sign.h>
+#endif
+
+#include "src/common/libutil/errprintf.h"
+
+#include "unwrap.h"
+#include "sign_none.h"
+
+char *unwrap_string_sign_none (const char *s,
+                               bool verify,
+                               uint32_t *userid,
+                               flux_error_t *errp)
+{
+    char *result = NULL;
+    int len;
+    uint32_t uid;
+    void *data = NULL;
+    if (sign_none_unwrap (s, (void **) &data, &len, &uid) < 0) {
+        errprintf (errp, "sign-none-unwrap failed: %s", strerror (errno));
+        return NULL;
+    }
+    if (verify && uid != getuid ()) {
+        errprintf (errp,
+                   "sign-none-unwrap: signing userid %lu != current %lu",
+                   (unsigned long) uid,
+                   (unsigned long) getuid ());
+        free (data);
```

Comment:
free potentially clobbers errno and I _think_ functions that call this one might expect errno to be valid (e.g. in the shell).

Reply:
Hm, I tried to explicitly _not_ depend on errno here (since what detail can errno give in this case), e.g. the comment on `unwrap_string()`:

```c
 *  If verify is true, then fail if signing mechanism is invalid or
 *   signing user does not match current uid. On failure, error.text
 *   is filled in with an error message. (errno not necessarily
 *   guaranteed to be valid).
```

The job shell uses `shell_log_error()` which logs an error but doesn't call `strerror()` (as `shell_log_errno()` would I believe).

Did you see a place where errno is expected to be valid?

Reply:
My mistake, sorry.  I saw `shell_log_error()` and assumed.  Also I didn't read your inline doc carefully.  Never mind!

Reply:
Yeah, unfortunate name collision between `shell_log_error()` and `log_error()`